src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (!(target instanceof Herbivore)) {
+      return;
+    }
+
+    if (!target.hidden) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        Animal.alive = Animal.alive.filter((animal) => animal !== target);
```

Comment:
The logic to remove a dead animal from the `Animal.alive` array is correct. However, be aware that directly modifying the array during iteration (which `filter` does internally) can lead to hard-to-track bugs. Since you're already inside the `Carnivore` class, consider using a method to handle the removal of dead animals to encapsulate this logic.